### PR TITLE
Document `IfNode` fields

### PR DESCRIPTION
**Pull request by dgollahon**
- Adds documentation for the fields of the `IfNode`. Also updates the overall description to mention ternary expressions.
- Part of #2123

### CODE REVIEW

**Comment by dgollahon** [comment]
Open to feedback on level of detail or any stylistic changes. I just opened this one node to see if it was approximately what you were looking for but happy to open others as a follow-up.

I also wasn't exactly sure how to indicate multiple lines for something like: 

```ruby
              if foo
                bar
              elsif baz
              ^^^^^^^^^
                qux
                ^^^
              end
              ^^^
```

so let me know if you have an opinion on that. Are the contents of this file used to generate documentation somewhere that I can see? I wasn't sure how to preview my changes outside of the YAML.

File: config.yml [hunk 1]
```diff
@@ -1554,26 +1554,84 @@ nodes:
...
+
+              bar if foo
+                     ^^^
+
+              foo ? bar : baz
+              ^^^
       - name: then_keyword_loc
         type: location?
+        comment: |
+          The location of the `then` keyword if present, `nil` otherwise.
```
Comment (kddnewton) [inline config.yml:1582]: This can also be `?` in a ternary.

**Review by kddnewton — CHANGES_REQUESTED** [review]
Great work, one small comment

File: config.yml (hunk 1 above)
Comment (dgollahon) [reply config.yml]: Updated, ready for another review @kddnewton. Sorry for the delay, I forgot I hadn't updated this.

**Review by kddnewton — APPROVED** [review]

**Pull request merged**